app/forms/application_form.rb

```diff
@@ -0,0 +1,26 @@
+class ApplicationForm
+  include ActiveModel::Model
+  include ActiveModel::Naming
+
+  validates :body, :user_id, presence: true
+
+  def save
+    update_form_attributes
+  end
+
+  def update
+    @attachments[:_destroy] ? del_attachment : update_form_attributes
```

Comment:
Looks like a logical mistake you or destroy attachment or update attribute isn't to do that at the same time?